src/Raven.Server/Documents/Handlers/QueriesHandler.cs

```diff
@@ -168,7 +168,7 @@ await using (var writer = new AsyncBlittableJsonTextWriter(queryContext.Document
             }
 
             Database.QueryMetadataCache.MaybeAddToCache(indexQuery.Metadata, result.IndexName);
-            AddPagingPerformanceHint(PagingOperationType.Queries, $"{nameof(Query)} ({result.IndexName})", indexQuery.Query, numberOfResults, indexQuery.PageSize, result.DurationInMs, totalDocumentsSizeInBytes);
+            AddPagingPerformanceHint(PagingOperationType.Queries, $"{nameof(Query)} ({result.IndexName})", $"{indexQuery.Metadata.QueryText}\n{indexQuery.QueryParameters}", numberOfResults, indexQuery.PageSize, result.DurationInMs, totalDocumentsSizeInBytes);
```

Comment:
Do we have to think about privacy concerns here?

Reply:
We are showing the same thing in traffic watch anyway.

Reply:
Ok

Reply:
Shouldn't we use `Environment.NewLine` instead of `\n` ? Not sure how Studio deals with that - @ml054 ?

Reply:
let's have a look at traffic watch - it works there correctly + it is property handled by studio. Usually we normalize new line endings.

Reply:
Note that for each query, even if performance hint is not added you are doing huge string concat

Reply:
@ppekrol Good point! 

I think we should extract this check to `ShouldAddPagingPerformanceHint()`:

https://github.com/ravendb/ravendb/blob/469a57ca47c39230a2c59d761cece9483e6761c9/src/Raven.Server/Documents/DatabaseRequestHandler.cs#L150-L151

and check it here before calling `AddPagingPerformanceHint()`

Reply:
or just wrap it up in some struct that will do the concat when if is passed

Reply:
After talking with Arek, we decided to pass a string as a delegate so concatenation is made only when we add a hint.